doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,682 @@
+---
+title: "Introduction to igraph"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Introduction to igraph}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Tutorial
+
+The main goals of the igraph library is to provide a set of data types and functions for 1) pain-free implementation of graph algorithms, 2) fast handling of large graphs, with millions of vertices and edges, 3) allowing rapid prototyping via high level languages like R.
+
+```{=html}
+<!---
+I take this introducction from the R tutorial in https://igraph.org/r/html/latest/aaa-igraph-package.html
+-->
+```
+
+## Creating a graph
+
+The simplest way to create a graph is the `graph` constructor. To make an empty graph:
+
+```{r}
+library("igraph")
+
+g <- make_empty_graph()
+```
+
+To make a graph with 10 nodes (numbered `1` to `10`) and two edges connecting nodes `1-2` and `1-5`:
+
+```{r}
+g <- make_graph(edges = c(1,2, 1,5), n=10, directed = FALSE)
+```
+
+We can print the graph to get a summary of its nodes and edges:
+
+```{r echo = TRUE}
+g
+```
+
+This means: **U** ndirected graph with **10** vertices and **2** edges, with the exact edges listed out. If the graph has a [name] attribute, it is printed as well.
+
+Note
+
+`summary` does not list the edges, which is convenient for large graphs with millions of edges:
+
+```{r echo = TRUE}
+summary(g)
+```
+
+`make_graph` can create some notable graphs. For example you can create a Frucht graph, who is the smallest cubical graph whose automorphism group consists only of the identity element. It has 12 vertices and 18 edges:
```

Comment:
I notice that throughout this text, you use "who" to refer to things. In English, we only use "who" to refer to people. Use "which" (or "that") when you are talking about things.

We say "cubic graph", not "cubical graph". When you're unsure about what the correct term is, do a google search and see how many results there are.

> whose automorphism group consists only of the identity element

This sounds highly technical, and many readers won't understand what it means. You could say "smallest asymmetric graph", or even "smallest graph without any non-trivial automorphisms", both of which are reasonable accurate mathematically, while also being somewhat understandable to readers who are not deeply familiar with graph theory. One can imagine what "asymmetric" might mean even without knowing the technical definition of "asymmetric" graphs.

To be perfectly accurate, the Frucht graph is not _the_ unique smallest asymmetric cubic graph, but one of five such graphs, all having the same size.

All this said, you don't necessarily need to define what the Frucht graph is to use it for demonstrations. If you do describe its properties, it would make sense to demonstrate them: compute the size of the automorphism group, and show the degrees to demonstrate that it is cubic.

Reply:
Thank you @szhorvat.

I'll take care of the English at the end, no need to worry about that quite yet.

I agree with you that this is perhaps a too technical example for the type of demonstration we have in mind at this stage. We'll just use something simpler (e.g. the karate club) to give people an idea without overwhelming the non-mathy reader.